src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -808,13 +831,24 @@ private void DecodeModuleInitializerAttribute(DecodeWellKnownAttributeArguments<
                 hasError = true;
             }
 
+            // In DecodeUnmanagedCallersOnlyAttribute, we check to see if this is a module initializer. In
+            // DecodeModuleInitializerAttribute, we check to see if this method is attributed with UnmanagedCallersOnly.
+            // When decoding a method we could have either order, so the first check will not find the presence of the
+            // other. The second check will find the presence and appropriately report an error.
+            if (arguments.GetOrCreateData<MethodWellKnownAttributeData>().UnmanagedCallersOnlyAttributeData is { } methodData)
+            {
+                Debug.Assert(!ReferenceEquals(methodData, UnmanagedCallersOnlyAttributeData.Uninitialized));
+                arguments.Diagnostics.Add(ErrorCode.ERR_ModuleInitializerCannotBeUnmanagedCallersOnly, arguments.AttributeSyntaxOpt.Location);
```

Comment:
>ERR_ModuleInitializerCannotBeUnmanagedCallersOnly [](start = 52, length = 49)

Why can't a module initializer be marked `[UnmanagedCallersOnly]`?

Reply:
Same reason `Main` can't be. It cannot be called from managed code after being marked with `UnmanagedCallersOnly`.

---
In reply to: [482328856](https://github.com/dotnet/roslyn/pull/47311#discussion_r482328856) [](ancestors = 482328856)